libraries/AP_AHRS/AP_AHRS.h

```diff
@@ -461,7 +461,7 @@ class AP_AHRS
     // set the EKF's origin location in 10e7 degrees.  This should only
     // be called when the EKF has no absolute position reference (i.e. GPS)
     // from which to decide the origin on its own
-    virtual bool set_origin(const Location &loc) WARN_IF_UNUSED { return false; }
+    virtual bool set_origin(const Location &loc) WARN_IF_UNUSED { WITH_SEMAPHORE(_rsem); return false; }
```

Comment:
No need to take a semaphore for a return false

Reply:
yes, I was wondering about this and just realized I got a bit confused with definitions! it should probably be in AP_AHRS_NavEKF::set_origin and not the header right? Excuse any obvious erros, diving into the deep end head first here!

Reply:
I've pushed changes - https://github.com/ArduPilot/ardupilot/pull/17766/commits/3e9a9a9e49237ad94bdefc87f37c7498a429e745

Hopefully this is better!